docs/pages/ready-made-features/ai-copilots/styling-and-customization.mdx

```diff
@@ -0,0 +1,182 @@
+for localization.
+
+## Default components
+
+To add the default components’ theme, import the
+[default styles](/docs/api-reference/liveblocks-react-ui#Default-styles) CSS
+file.
+
+```ts
+import "@liveblocks/react-ui/styles.css";
+```
+
+You can also import one of two CSS files to enable
+[dark mode](/docs/api-reference/liveblocks-react-ui#Dark-mode), depending on how
+you’d like to enable it.
+
+```ts
+// Dark mode using the system theme with `prefers-color-scheme`
+import "@liveblocks/react-ui/styles/dark/media-query.css";
+```
+
+```ts
+// Dark mode using `className="dark"`, `data-theme="dark"`, or `data-dark="true"`
+import "@liveblocks/react-ui/styles/dark/attributes.css";
+```
+
+### CSS variables
+
+A number of
+[CSS variables](/docs/api-reference/liveblocks-react-ui#CSS-variables) can be
+used to customize colors, spacing, and more. This is our recommended path for
+styling the default components, as you can quickly and easily modify all
+components with just a few variables.
+
+```css
+/* Styles all default components */
+.lb-root {
+  --lb-accent: purple;
+  --lb-spacing: 1em;
+  --lb-radius: 0;
+}
+```
+
+### Class names
+
+Should you need deeper customization,
+[class names](/docs/api-reference/liveblocks-react-ui#Class-names) can be
+styled, some of which provide contextual data attributes.
+
+```css
+.lb-ai-chat {
+  /* Customize AI chat */
+}
+
+.lb-ai-composer {
+  /* Customize message composer */
+}
+```
+
+### Overrides and localization
+
+It’s possible to
+[override strings](/docs/api-reference/liveblocks-react-ui#Overrides) used in
+the default components, which has a couple of different uses, the first being
+localization. In this example, we’re globally setting the AI composer’s
+placeholder from “Ask anything…” to “Posez une question” for French users.
+
+```tsx
+import { LiveblocksUiConfig } from "@liveblocks/react-ui";
+
+export function App() {
+  return (
+    <LiveblocksUiConfig
+      overrides={{
+        // +++
+        locale: "fr",
+        AI_COMPOSER_PLACEHOLDER: "Posez une question",
+        // +++
+      }}
+    >
+      {/* ... */}
+    </LiveblocksUiConfig>
+  );
+}
+```
+
+You can also override strings on a component basis, for example if you’d like to
+change the “Send” button tooltip to “Post message” in the chat’s message
+composer.
+
+```tsx
+import { Composer } from "@liveblocks/react-ui";
+
+function Component() {
+  return (
+    <AiChat
+      chatId="my-chat-id"
+      copilotId="co_a7Gd3x..."
+      overrides={{
+        // +++
+        AI_COMPOSER_PLACEHOLDER: "Post message",
+        // +++
+      }}
+    />
+  );
+}
+```
+
+## Custom interfaces
+
+You can bypass the default components and build completely custom UI for your AI
+copilots, for example using
+[`useAiChatMessages`](/docs/api-reference/liveblocks-react#useAiChatMessages)
+and [`useSendAiMessage`](/docs/api-reference/liveblocks-react#useSendAiMessage)
+to build your own chat history and composer.
+
+```tsx
```

Comment:
I know that we haven't made primitives public yet (even though they're 90+% ready, but we're uncertain about them blocking us later) to make this easier, but if you do this in the meantime it will prevent tools (tool invocations) to run, `execute` callbacks won't be triggered. cc: @nvie